Cold start of Notify SDK with History API (#796)
WalletConnect/walletconnect-docs: docs/specs/clients/push/usage-of-history-api.md

# Cold start of Notify SDK with History API

## Motivation

> A cold start means that Push SDK running for the first time on a device without any cached state. The account that is used for `enableSync` of PushClient API may already have some kind of history (pushSubscriptions, pushMessages). The HistoryAPI's purpose is to restore history and state after a cold start.

Reviewer: ```suggestion
A cold start represents the Push SDK running for the first time on a device without any cached state. The account that is used for `enableSync` of PushClient API may already have some kind of history (pushSubscriptions, pushMessages). The HistoryAPI's purpose is to restore history and state after a cold start.
```

> A cold start means that Push SDK running the first time on a device without any cached state. Account that is using for `enableSync` of PushClient API may already have some kind of history (pushSubscriptions, pushMessages). The purpose of using the HistoryAPI is to restore history and state after a cold start.

Reviewer 2: ```suggestion
A cold start means that Push SDK running for the first time on a device without any cached state. The account that is used for `enableSync` of PushClient API may already have some kind of history (pushSubscriptions, pushMessages). The HistoryAPI's purpose is to restore history and state after a cold start.
```

Note: Launching a client after 30 days of inactivity is also considered a cold start

## Implementation

> 1. W1 already have some `pushSubscriptions`, `pushMessages` history

Reviewer: ```suggestion
1. W1 already has some `pushSubscriptions`, `pushMessages` history
```

> 2. Wn performing cold start on different device and have same account as W1

Reviewer 2: ```suggestion
2. Wn performing cold start on different device and have the same account as W1
```

> 2. Wn performing cold start on different device and have the same account as W1

Reviewer: ```suggestion
2. Wn performs cold start on different device but the same account as W1
```

> 3. Wn configuring sync for account with `enableSync` method of PushClient

Reviewer 2: ```suggestion
3. Wn configuring sync for an account with `enableSync` method of PushClient
```

> 3. Wn configuring sync for an account with `enableSync` method of PushClient

Reviewer: ```suggestion
3. Wn configures sync for an account with `enableSync` method of PushClient
```

> 4. Wn registers HistoryAPI for `wc_syncSet` and `wc_syncDel` SyncAPI methods. Request tags can be found in [Sync API](../core/sync/readme.md) specs

Reviewer 3: Needs to also register `push_message`

Author: updated

> 4. Wn register HistoryAPI for `wc_syncSet` and `wc_syncDel` SyncAPI methods. Request tags could found in [Sync API](../core/sync/readme.md) specs

Reviewer: ```suggestion
4. Wn registers HistoryAPI for `wc_syncSet` and `wc_syncDel` SyncAPI methods. Request tags can be found in [Sync API](../core/sync/readme.md) specs
```

> 5. Wn fetching sync messages containing PushSubscription payloads. Topic to fetch and keys to decrypt payloads is deriving accoring [Sync API](../core/sync/readme.md). PushSubscriptions description could found in[PushSubscriptions sync storage specs](./usage-of-sync-api.md)

Reviewer 2: ```suggestion
5. Wn fetching sync messages containing PushSubscription payloads. Topic to fetch and keys to decrypt payloads is deriving according [Sync API](../core/sync/readme.md). PushSubscriptions description could found in[PushSubscriptions sync storage specs](./usage-of-sync-api.md)
```

> 5. Wn fetching sync messages containing PushSubscription payloads. Topic to fetch and keys to decrypt payloads is deriving according [Sync API](../core/sync/readme.md). PushSubscriptions description could found in[PushSubscriptions sync storage specs](./usage-of-sync-api.md)

Reviewer: ```suggestion
5. Wn fetches sync messages containing PushSubscription payloads. Topic to fetch and keys to decrypt payloads are derived according to [Sync API](../core/sync/readme.md). PushSubscriptions description can be found in [PushSubscriptions sync storage specs](./usage-of-sync-api.md)
```

> 6. Wn combines `pushSubscriptions` inserts with deletions and updating local database

Reviewer 2: ```suggestion
6. Wn combines `pushSubscriptions` inserts with deletions and updates the local database
```

> 7. Wn subscribing for `pushSubscriptions` topics

Reviewer: ```suggestion
7. Wn subscribes for `pushSubscriptions` topics
```

> 8. Wn setting subscription symKey P into keychain to be able to decrypt pushMessages payloads

Reviewer: ```suggestion
8. Wn sets subscription symKey P in keychain to be able to decrypt pushMessages payloads
```

> 9. Wn fetching `wc_pushMessage` messages for every subscription topic from HistoryAPI. Request tags could found in [Push RPC methods](./rpc-methods.md) 

Reviewer: ```suggestion
9. Wn fetches `wc_pushMessage` messages for every subscription topic from HistoryAPI. Request tags can be found in [Push RPC methods](./rpc-methods.md) 
```

> 10. Wn decrypt pushMessages payload with symKey P and updating local database

Reviewer 2: ```suggestion
10. Wn decrypts pushMessages payload with symKey P and updates local database
```

# Cold start of Notify SDK with History API

## Motivation

> A cold start means that Push SDK running the first time on a device without any cached state. Account that is using for `enableSync` of PushClient API may already have some kind of history (pushSubscriptions, pushMessages). The purpose of using the HistoryAPI is to restore history and state after a cold start.

Author: If client inactive more than 30 days == cold start

## Implementation

1. W1 already have some `pushSubscriptions`, `pushMessages` history
2. Wn performing cold start on different device and have same account as W1
3. Wn configuring sync for account with `enableSync` method of PushClient
4. Wn register HistoryAPI for `wc_syncSet` and `wc_syncDel` SyncAPI methods. Request tags could found in [Sync API](../core/sync/readme.md) specs
5. Wn fetching sync messages containing PushSubscription payloads. Topic to fetch and keys to decrypt payloads is deriving accoring [Sync API](../core/sync/readme.md). PushSubscriptions description could found in[PushSubscriptions sync storage specs](./usage-of-sync-api.md)
6. Wn combines `pushSubscriptions` inserts with deletions and updating local database
7. Wn subscribing for `pushSubscriptions` topics
8. Wn setting subscription symKey P into keychain to be able to decrypt pushMessages payloads
9. Wn fetching `wc_pushMessage` messages for every subscription topic from HistoryAPI. Request tags could found in [Push RPC methods](./rpc-methods.md) 

> 10. Wn decrypt pusMessages payload with symKey P and updating local database

Author: pushMessages

To configure HistoryAPI client should register tags that will be tracked by History server. 

```swift

> 	History.instance.register(tags: ["5000", "5002"])

Author: "4002" for pushMessages

Author: -> js
